examples/src/CustomDateRange.tsx

```diff
@@ -54,15 +56,31 @@ export class Index extends React.Component<any, any> {
                     startDate={this.state.startDate}
                     endDate={this.state.endDate}
                     onSave={this.onSave}
-                    onClose={this.toggleDialog}>
+                    onClose={this.toggleDialog}
+                    onDateSelectionChanged={this.onDateSelectionChanged}
+                    invalidDateRangeSelected={this.state.invalidDateRangeSelected}
+                    invalidErrorMessage={this.state.invalidErrorMessage}>
                 </CustomDateRange>
             </div>
         );
     }
 
     @autobind
     private toggleDialog() {
-        this.setState({ isOpen: !this.state.isOpen });
+        this.setState({ 
+            isOpen: !this.state.isOpen,
+            invalidDateRangeSelected: false,
+            invalidErrorMessage: ''
+        });
+    }
+
+    @autobind
+    private onDateSelectionChanged(selectedStartDate: Date, selectedEndDate: Date) {
+        const invalidSelectedDate = moment(selectedStartDate) <= moment(selectedEndDate).subtract(3, 'day');
+        this.setState({
+            invalidDateRangeSelected: invalidSelectedDate,
+            invalidErrorMessage: invalidSelectedDate ? 'The time period between Start and End date must not be more than 3 days.' : ''
```

Comment:
This seems a bit specific? Can we take the same approach as in TextField component, to use invalidErrorMessage prop to define invalid date selection?